app.rb

```diff
@@ -0,0 +1,171 @@
+require_relative 'person'
+require_relative 'book'
+require_relative 'student'
+require_relative 'classroom'
+require_relative 'rental'
+require_relative 'teacher'
+
+class App
+  def initialize
+    @books = []
+    @students = []
+    @teachers = []
+    @people = []
+    @rentals = []
+  end
+
+  # list of books
+  def list_books
+    if @books.empty?
+      puts 'No books found.'
+    else
+      @books.each_with_index do |book, index|
+        puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
+      end
+    end
+    nil
+  end
+
+  # list of people method
+  def list_people
+    print 'Do you want to Create a list of a students (1) or a teachers (2)? [Input the number]: '
+    list_person = gets.chomp.to_i
+    case list_person
+    when 1
+      list_student
+    when 2
+      list_teacher
+    else
+      puts "\nInvalid input!!\n\n"
+    end
+  end
+
+  def list_student
+    if @students.empty?
+      puts 'No students found.'
+    else
+      @students.each_with_index do |student, index|
+        puts "#{index}) [#{student.class}] ID: #{student.id}, Name: #{student.name},
+         Age: #{student.age}, classroom: #{student.classroom}"
+      end
+    end
+    nil
+  end
+
+  def list_teacher
+    if @teachers.empty?
+      puts 'No teachers found.'
+    else
+      @teachers.each_with_index do |teacher, index|
+        puts "#{index}) [#{teacher.class}] ID: #{teacher.id}, Name: #{teacher.name},
+         Age: #{teacher.age}, specialization: #{teacher.specialization}"
+      end
+    end
+    nil
+  end
+
+  # Create a person method
+  def create_person
+    print 'Do you want to Create a student (1) or a teacher (2)? [Input the number]: '
+    select_person = gets.chomp.to_i
+    case select_person
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts "\nInvalid input\n\n"
+      create_person
+    end
+  end
+
+  def create_student
+    print 'Classroom: '
+    student_classroom = gets.chomp
+    print 'Age: '
+    student_age = gets.chomp.to_i
+    print 'Name: '
+    student_name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp
+    student = Student.new(student_classroom, student_age, student_name, parent_permission: parent_permission)
```

Comment:
- [ ] you have added the parent_permission wrongly here please see the code snippet below for how it should be used :+1: 
- code snippet
```ruby 
    student = Student.new(student_classroom, student_age, student_name, parent_permission)
```